openlibrary/plugins/worksearch/code.py

```diff
@@ -636,16 +636,59 @@ class list_search(delegate.page):
     path = '/search/lists'
 
     def GET(self):
+        i = web.input(q='', offset='0', limit='10')
+
+        lists = self.get_results(i.q, i.offset, i.limit)['response']['docs']
+
+        return render_template('search/lists.tmpl', q=i.q, lists=lists)
+
+    def get_results(self, q, offset=0, limit=100):
         if 'env' not in web.ctx:
             delegate.fakeload()
 
-        i = web.input(q='', offset='0', limit='10')
         keys = web.ctx.site.things({
-            "type": "/type/list", "name~": i.q,
-            "limit": int(i.limit), "offset": int(i.offset)
+            "type": "/type/list", "name~": q,
+            "limit": int(limit), "offset": int(offset)
         })
         lists = web.ctx.site.get_many(keys)
-        return render_template('search/lists.tmpl', q=i.q, lists=lists)
+
+        return {
+            'response': {
+                'docs': lists
+            }
+        }
+
+class list_search_json(list_search):
+    path = '/search/lists'
+    encoding = 'json'
+
+    def GET(self):
+        i = web.input(q='', offset=0, limit=10)
+        offset = safeint(i.offset, 0)
+        limit = safeint(i.limit, 10)
+        limit = min(100, limit)  # limit limit to 1000.
+
+        response = self.get_results(i.q, offset=offset, limit=limit)['response']
+
+        response['docs'] = [
+            {
+                'key': list.key,
+                'url': list.url(),
+                'name': list.name,
+                'description': str(list.description),
+                'last_modified': list.last_modified.isoformat() if list.last_modified else None,
+                'seed_count': len(list.seeds),
+                'edition_count': list.edition_count,
+                'owner': {
+                    'key': list.get_owner().key,
+                },
+                'top_subjects': [{"key": s.key, "url": s.url, "name": s.name} for s in list.get_top_subjects(limit=5)],
+            }
```

Comment:
Let's put this in a helper method. The type of these is `List` (see https://github.com/internetarchive/openlibrary/blob/f7257116e513631f42f4f8f03a8eb31e252b9044/openlibrary/core/models.py#L699 ). It looks like there's a similar function (called `preview`) in the `ListMixin` superclass, but I think we should create a new one. You'll then be able to call it directly on `list` (similar to how `get_top_subjects` is called).

Reply:
Can do - that would make the code simpler if we needed to include more author information anyway!

Reply:
![Screenshot from 2019-10-10 12-18-14](https://user-images.githubusercontent.com/1551593/66587323-2d3afa00-eb58-11e9-813f-ffd02100cd02.png)

Reply:
:+1: Seems good; it's always easy to add fields later; not so easy to remove them :P